src/Authoring/WinRT.SourceGenerator/AotOptimizer.cs

```diff
@@ -1102,6 +1103,19 @@ private static bool NeedVtableOnLookupTable(SyntaxNode node)
                     }
                 }
             }
+            // Detect scenarios where the property declaration has an initializer and is to a boxed or cast type during initialization.
+            else if (context.Node is PropertyDeclarationSyntax propertyDeclaration)
+            {
+                var leftSymbol = context.SemanticModel.GetSymbolInfo(propertyDeclaration.Type).Symbol;
+                if (propertyDeclaration.Initializer != null)
+                {
+                    var instantiatedType = context.SemanticModel.GetTypeInfo(propertyDeclaration.Initializer.Value);
+                    if (leftSymbol is INamedTypeSymbol namedType)
+                    {
+                        AddVtableAttributesForType(instantiatedType, namedType);

```

Comment:
Do we need any validation on `instantiatedType`? Or will `AddVtableAttributesForType` take care of that?

Reply:
`AddVtableAttributesForType` will validate it.